### PR TITLE
fix(core): stop minting typed relations from prose tails

### PR DESCRIPTION
**Pull request by phernandez**
## Why

`- Added [[Target Note]] to the roster` minted a typed relation `Added`, and the tail `to the roster` — including any further `[[links]]` in it — was silently dropped from the edge. Measured at 12.3% of typed edges being parse artifacts in a real vault (#1260).

## What changed

Grammar policy (direction 1 from the issue): **an explicit relation line ends at its target or its `(context)`**. Any other trailing text means the line is a sentence, and it falls through to inline `links_to` handling — which also preserves every wikilink in the tail (the old explicit path lost `[[Beta]]` entirely in `- relates_to [[Alpha]] and [[Beta]]`).

Preserved: `- spouse_of [[X]]`, `- requires [[X]] (context)`, quoted multi-word types, `- 使用 [[X]]`. Changed deliberately: unclosed-context and nested-bracket edge cases now fall back to inline links instead of minting a typed edge with a mangled tail.

**Known limitation, pinned in tests**: `- Mother [[Target Note]]` (capitalized single word, no tail) still mints — indistinguishable by shape from a hand-authored `Requires [[X]]`. The policy for that case (lowercase requirement / structural anchoring) stays open for discussion on #1260.

## Verification

- tests/markdown: 88 passed (4 new tests: prose-tail fallback, tail-link preservation, preserved explicit forms, pinned limitation)
- tests/services + tests/mcp: 1316 passed, 2 skipped
- ruff format, ruff check, ty: clean

Refs #1260 (left open for the remaining grammar-policy discussion).

Credit: analysis, repro, and the measured vault impact are @the issue reporter's — see #1260.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_01G4rbaeHJN3L7CREp5v38J9

### CODE REVIEW

File: src/basic_memory/markdown/plugins.py [hunk 1]
```diff
@@ -145,41 +145,51 @@ def is_explicit_relation(token: Token) -> bool:
...
+    #   tail means the line is a sentence that happens to contain a wikilink; the
+    #   old behavior minted a junk type from the word before the link and silently
+    #   dropped the tail — including any further [[links]] in it — from the edge
+    #   (#1260).
+    # Outcome: such lines fall through to inline links_to handling, which keeps
+    #   every wikilink on the line as an edge and the sentence intact as content.
+    after = content[end + 2 :].strip()
+    context = None
+    if after.startswith("(") and after.endswith(")"):
+        context = after[1:-1].strip() or None
```
[the remark this answers is not in the document]
Comment (phernandez) [reply src/basic_memory/markdown/plugins.py]: Fixed in e70713b6: _is_single_parenthesized requires the opening paren to close exactly at the final character (depth-tracked), so '(primary) and [[Beta]] (secondary)' is a prose tail and falls back to inline links — regression asserts both Alpha and Beta survive as links_to. '(with (nested) parens)' stays a valid single context. 88 markdown tests pass.

**Comment by phernandez** [comment]
@codex review

File: src/basic_memory/markdown/plugins.py [hunk 2]
```diff
@@ -145,41 +145,72 @@ def is_explicit_relation(token: Token) -> bool:
...
+    if not text.startswith("("):
+        return False
+    depth = 0
+    for position, char in enumerate(text):
+        if char == "(":
+            depth += 1
+        elif char == ")":
+            depth -= 1
+            if depth == 0:
+                return position == len(text) - 1
```
[the remark this answers is not in the document]
Comment (phernandez) [reply src/basic_memory/markdown/plugins.py:203]: Declining as conservative-by-design. The relation grammar processes raw text and has never been escape- or inline-code-aware anywhere (types, targets, contexts); teaching only the paren balancer about Markdown literals adds grammar complexity for an exotic authoring shape. The failure direction is safe: the line keeps its wikilink as a links_to edge and its full text as content — nothing is minted wrong and nothing is lost, which is the corruption class this PR exists to prevent. Pre-#1291 behavior kept the type for these shapes but was the same lenient check that silently ate links. If real vaults surface escaped-paren contexts, that belongs in the #1260 grammar-policy discussion (v0.23.1).
